docs/src/man/getting_started.md

```diff
@@ -118,68 +162,223 @@ but for very large `DataFrame`s  this may be a consideration.
-│ 1   │ 1 │ M │
-│ 2   │ 2 │ F │
-│ 3   │ 3 │ F │
-│ 4   │ 4 │ M │
-│ 5   │ 5 │ F │
-│ 6   │ 6 │ M │
+6×3 DataFrame
+│ Row │ A     │ B     │ C     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 1     │ 1     │ 1     │
+│ 2   │ 3     │ 1     │ 2     │
+│ 3   │ 5     │ 1     │ 3     │
+│ 4   │ 7     │ 1     │ 4     │
+│ 5   │ 9     │ 1     │ 5     │
+│ 6   │ 11    │ 1     │ 6     │
 
 julia> tail(df)
-6×2 DataFrames.DataFrame
-│ Row │ A │ B │
-├─────┼───┼───┤
-│ 1   │ 3 │ F │
-│ 2   │ 4 │ M │
-│ 3   │ 5 │ F │
-│ 4   │ 6 │ M │
-│ 5   │ 7 │ M │
-│ 6   │ 8 │ F │
+6×3 DataFrame
+│ Row │ A     │ B     │ C     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 989   │ 10    │ 495   │
+│ 2   │ 991   │ 10    │ 496   │
+│ 3   │ 993   │ 10    │ 497   │
+│ 4   │ 995   │ 10    │ 498   │
+│ 5   │ 997   │ 10    │ 499   │
+│ 6   │ 999   │ 10    │ 500   │
+```
+
+### Taking a Subset
 
+Specific subsets of a data frame can be extracted using the indexing syntax, similar to matrices. The colon `:` indicates that all items (rows or columns depending on its position) should be retained:
+
+```jldoctest dataframe
 julia> df[1:3, :]
-3×2 DataFrames.DataFrame
-│ Row │ A │ B │
-├─────┼───┼───┤
-│ 1   │ 1 │ M │
-│ 2   │ 2 │ F │
-│ 3   │ 3 │ F │
+3×3 DataFrame
+│ Row │ A     │ B     │ C     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 1     │ 1     │ 1     │
+│ 2   │ 3     │ 1     │ 2     │
+│ 3   │ 5     │ 1     │ 3     │
+
+julia> df[[1, 5, 10], :]
+3×3 DataFrame
+│ Row │ A     │ B     │ C     │
+│     │ Int64 │ Int64 │ Int64 │
+├─────┼───────┼───────┼───────┤
+│ 1   │ 1     │ 1     │ 1     │
+│ 2   │ 9     │ 1     │ 5     │
+│ 3   │ 19    │ 1     │ 10    │
+
+julia> df[:, [:A, :B]]
+500×2 DataFrame
+│ Row │ A     │ B     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 1     │
+│ 2   │ 3     │ 1     │
+│ 3   │ 5     │ 1     │
+│ 4   │ 7     │ 1     │
+⋮
+│ 496 │ 991   │ 10    │
+│ 497 │ 993   │ 10    │
+│ 498 │ 995   │ 10    │
+│ 499 │ 997   │ 10    │
+│ 500 │ 999   │ 10    │
+
+julia> df[1:3, [:B, :A]]
+3×2 DataFrame
+│ Row │ B     │ A     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 1     │
+│ 2   │ 1     │ 3     │
+│ 3   │ 1     │ 5     │
+
+julia> df[[3, 1], [:C]]
+2×1 DataFrame
+│ Row │ C     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 3     │
+│ 2   │ 1     │
+```
+
+Do note that `df[[:A]]` and `df[:, [:A]]` return a `DataFrame` object, while `df[:A]` and `df[:, :A]` return a vector:
+
+```jldoctest dataframe
+julia> df[[:A]]
+500×1 DataFrame
+│ Row │ A     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 1     │
+│ 2   │ 3     │
+│ 3   │ 5     │
+│ 4   │ 7     │
+⋮
+│ 496 │ 991   │
+│ 497 │ 993   │
+│ 498 │ 995   │
+│ 499 │ 997   │
+│ 500 │ 999   │
+
+julia> df[[:A]] == df[:, [:A]]
+true
 
+julia> df[:A]
+500-element Array{Int64,1}:
+   1
+   3
+   5
+   7
+   9
+  11
+   ⋮
+ 991
+ 993
+ 995
+ 997
+ 999
+
+julia> df[:A] === df[:, :A]
```

Comment:
This is going to be deprecated (also above), only `==` will pass in the future as we want to be consistent with indexing in Base, where `:` creates a copy.